app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/UndertakingController.scala

```diff
@@ -260,6 +283,17 @@ class UndertakingController @Inject()(
     )
   }
 
+  private def getLanguage(implicit request: EscAuthRequest[_]): Future[Language] = request.request.messages.lang.code.toLowerCase(Locale.UK) match {
+    case English.code => Future.successful(English)
+    case Welsh.code   => Future.successful(Welsh)
+    case other        => sys.error(s"Found unsupported language code $other")
+  }
+
+  private def getTemplateId(lang: Language) = lang match {
+    case English => templateIdEN
```

Comment:
If these `templateId` variables are only used here then you may as well inline them.

Reply:
Just for future proofing , we may have to call this at some where else. Plus i think it makes code cleaner. what do you think?

Reply:
Fair enough. :)